### PR TITLE
feat(VCalendar): general improvements & category view

### LINKED ISSUE #8667: [Feature Request] A 'initial time props' to Daily and Weekly v-calendar
### Problem to solve
To indicate a initial time on a Daily or Weekly calendar, we have to setup the 'first' block number according to the 'minutes' number. 
For example, if we want to setup a calendar that starts at 2:00 but the 'minutes' value is 45, we have to set the 'first' value with 2 (because the block starts count at 00:00 with the block index equal to 0), this block will be equivalent to 1:30 and the block 3 will be equivalent to 2:15.  
That’s a calendar deficit because we need to count the exactly block that indicates the time that we want to setup, and according with the 'minutes' value we don't have our exactly initial time setup.

### Reproduction Link
https://codepen.io/larissagilliane/pen/mdbOjme

### Proposed solution
Add an option that set the initial time, not the first block.
If i want to start my calendar at 2:00 and i have my 'minutes' as 45. i need to set my initial time at 2:00 and the block will evolve to 2:45, 3:30 and successively.

### LINKED ISSUE #11080: [Feature Request] Add more scheduling features to calendar
### Problem to solve
I would love it if we could add more scheduling features in the calendar. Honestly just two features would suffice.

1. Add resource view with daily appointments, something vertical like
      Resource1   Resource2
    10am
    11am
    12pm

2. Then the ability to click into a cell and get the context of the resource as well as the date/time
    

### Proposed solution
This solution isn't too far off from the week view, but instead of incrementing days, you would loop over resources and show the same day for each one. Then you would just add a click on cells and pass context. Honestly this shouldn't be a huge addon. The reason I'm asking for this is cause I've tried so many other schedulers, there's nothing really good out there for vue. There is syncfusion and bryntum but these are plain javascript libraries that wrap themselves in in popular frameworks like vue and angular, however these cause tons of issues when trying to customize them since they are not component based at core.

### LINKED ISSUE #11093: [Feature Request] v-calendar allow timestamp and Date to be passed as date values
### Problem to solve
Implementing a calendar which is timezone friendly is non-trivial with the current string requirements. Allowing date and timestamps to be passed will allow a timezone feature to be added later, and will also alleviate the need for the user to constantly format all their dates based on the current timezone. 

### Proposed solution
Update parseTimestamp to accept number and Date. Update prop validators and types. Make parseTimestamp and parseEvent more easily accessible as well.

### LINKED ISSUE #11133: [Feature Request] Improve event slot for VCalendar
### Problem to solve
You have access to the original event, but the parsed event is essential for common calendar functionality. It would also be nice to have access to functions which are used to display the event name and potentially a summary of the time.

### Proposed solution
Add to slot scope
- parsedEvent
- getEventName
- formatTime
- singleline

### LINKED ISSUE #11182: [Feature Request] v-calendar add title property to calendar
### Problem to solve
It's very common to summarize the time span you're looking at (specific Month, Week, Day, etc). Add a simple computed property to the title would make it easier for users and would simplify examples.

### Proposed solution
Add the computed title

### PR DESCRIPTION
**Pull request by ClickerMonkey**
## Description

- Allows millis or Dates as input instead of just formatted strings (resolves #11093)
- Improve event slot (resolves #11133)
- Added title computed value (resolves #11182)
- Added first-time prop (resolves #8667)
- Added category calendar type (resolves #11080)

## How Has This Been Tested?

unit & visually

## Markup:

You can find markup for each feature in the linked issues above

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by ClickerMonkey** [comment]
Category view previews:

1day2cat
1day2cat-slot
2day2cat-slot

File: packages/vuetify/src/components/VCalendar/util/props.ts [hunk 1]
```diff
@@ -109,10 +113,33 @@ export default {
...
+    },
+    categoryHideDynamic: {
+      type: Boolean,
+      default: false,
+    },
+    categoryShowAll: {
+      type: Boolean,
+      default: false,
+    },
+    categoryForInvalid: {
```
Comment (johnleider) [inline packages/vuetify/src/components/VCalendar/util/props.ts:133]: Strings are empty by default, `''`

File: packages/vuetify/src/components/VCalendar/util/props.ts [hunk 2]
```diff
@@ -8,12 +8,12 @@ import { PropValidator } from 'vue/types/options'
...
   base: {
     start: {
-      type: String,
+      type: [String, Number, Date],
       validate: validateTimestamp,
       default: () => parseDate(new Date()).date,
     },
     end: {
-      type: String,
+      type: [String, Number, Date],
```
[the remark this answers is not in the document]
Comment (jacekkarczmarczyk) [reply packages/vuetify/src/components/VCalendar/util/props.ts:16]: See for example https://github.com/vuetifyjs/vuetify/issues/1939 or https://github.com/vuetifyjs/vuetify/issues/2022, maybe they could be avoided but some of the problems were happening on apples and i don't have mac/iphone to be able to verify/fix them
Also manual testing would probably require using different time zones (or maybe even dates related to Daylight Saving Time) in system settings, no idea how to deal with it with jest testing.

File: packages/vuetify/src/components/VCalendar/util/props.ts (hunk 1 above)
Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/util/props.ts:133]: I did a simple test and they appear to be undefined

File: packages/vuetify/src/components/VCalendar/util/props.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ClickerMonkey) [reply packages/vuetify/src/components/VCalendar/util/props.ts:16]: Allowing dates greatly simplifies the code for the end-user - and I don't think we'll see any of the problems that were seen with the Date Picker. Users of the VCalendar were using Dates anyway then improperly converting them back and forth to strings. The Calendar still uses an internal VTimestamp object, the only thing that changes is taking a Date as input and it converts it to the VTimestamp the same way it did for determine the current date (the logic was already there, it's just being exposed for inputs). Real world uses of Calendars store event start and end dates in a timezone friendly format, like a UTC timestamp which interfaces perfectly with a JS Date, and terribly with a plain string.

The code in the VCalendar uses the same methods that were employed to fix the v-date-picker issue, the VTimestamp is essentially stored as a string which is formatted with UTC (using the createNativeFormatterXYZ function). Nothing is really changing under the hood.

[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VCalendar/util/props.ts:16]: If this is the current process for the calendar component, we can proceed as such. I would however like to set a standard of how we handle this use-case and possibly how we can abstract it for re-use in v3.

**Pull request merged**